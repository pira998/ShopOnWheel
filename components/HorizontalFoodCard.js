import React from "react";

import {
    TouchableOpacity,View,Text,Image
}from 'react-native';

import { COLORS,FONTS,SIZES,icons } from "../constants";


const HorizontalFoodCard =({containerStyle,imageStyle,item,onPress})=>{
    return(
        <TouchableOpacity
        style={{
            flexDirection:'row',
            borderRadius:SIZES.radius,
            backgroundColor:COLORS.lightGray2,
            ...containerStyle
        }}>

            {/*image */}
            <Image 
            source={item.image}
            style={imageStyle}
            />

            {/* info */}
            <View
            style={
                {
                    flex:1
                }
            }>
                    {/* Name*/}
                    <Text
                    style={
                        {
                            ...FONTS.h3,fontSize:17
                        }
                    }>{item.name}</Text>


                    {/* Discription */}
                    <Text
                    style={
                        {
                            ...FONTS.h3,fontSize:14,color:COLORS.darkGray
                        }
                    }>{item.description}</Text>


                    {/* price */}
                    <Text
                    style={
                        {
                            ...FONTS.h3,fontSize:17,marginTop:SIZES.base
                        }
                    }>${item.price}</Text>

            </View>


        </TouchableOpacity>

    )
}

export default HorizontalFoodCard;