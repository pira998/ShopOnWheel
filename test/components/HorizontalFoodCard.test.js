import React from 'react';
import renderer from 'react-test-renderer';
import HorizontalFoodCard from '../../components/HorizontalFoodCard'


test('renders correctly', () => {
  const item = {
      name:'testing',
      image:'testImage'
  }
  const tree = renderer.create(<HorizontalFoodCard item={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});