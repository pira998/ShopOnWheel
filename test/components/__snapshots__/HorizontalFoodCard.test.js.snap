// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#F5F5F8",
      "borderRadius": 12,
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "testImage",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "sans-serif",
          "fontSize": 17,
          "lineHeight": 22,
        }
      }
    >
      testing
    </Text>
    <Text
      style={
        Object {
          "color": "#525C67",
          "fontFamily": "sans-serif",
          "fontSize": 14,
          "lineHeight": 22,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "sans-serif",
          "fontSize": 17,
          "lineHeight": 22,
          "marginTop": 8,
        }
      }
    >
      $
    </Text>
  </View>
</View>
`;
